### PR TITLE
add cub w/ package config install

### PR DESCRIPTION
**Pull request by headupinclouds**
Notes
* this is a CUDA *.cuh header-only lib where the install stage simply copies the header tree.
* this release is hosted here https://github.com/headupinclouds/cub
* needs  https://github.com/hunter-packages/cub (see https://github.com/ruslo/hunter/issues/1157)

Testing would require a CUDA setup w/ nvcc toolchain (Possible TODO)

- [x] has hunter-packages/cub

### CODE REVIEW

**Comment by xsacha** [comment]
Thanks.
I've found cub keeps deleting history and removing release commits, which is annoying for checking out Caffe.
Do you know if we can just use the latest version of Cub for all versions of Caffe and Tensorflow?

**Comment by headupinclouds** [comment]
> Do you know if we can just use the latest version of Cub for all versions of Caffe and Tensorflow?

I'm not sure.  I guess we can try and see.  I think finding some CI options for testing this GPGPU/CUDA stuff will be needed.

TODO: grpc, boringssl, lmdv, sync, re2, snappy, jemalloc

**Comment by ruslo** [comment]
> needs https://github.com/hunter-packages/cub

Done: https://github.com/hunter-packages/cub

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2734
* https://travis-ci.org/ingenue/hunter/builds/300145146?utm_source=github_status&utm_medium=notification

File: examples/cub/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+# * https://github.com/hunter-packages/gate
+include("../common.cmake")
+
+project(download-cub)
+
+# Test CMake cub installation, even if find_package(CUDA) fails
+hunter_add_package(cub)
+find_package(cub CONFIG REQUIRED)
+
+find_package(CUDA QUIET REQUIRED)
```
Comment (ruslo) [inline examples/cub/CMakeLists.txt]: Got error:
```
CMake Error at /.../cmake-3.10/Modules/FindCUDA.cmake:682 (message):
  Specify CUDA_TOOLKIT_ROOT_DIR
Call Stack (most recent call first):
  CMakeLists.txt:17 (find_package)
```

I think for the start we can do just:
```cmake
hunter_add_package(cub)
find_package(cub CONFIG REQUIRED)

if(NOT TARGET cub::cub)
  message(FATAL_ERROR "Target not found")
endif()
```

Comment (headupinclouds) [reply examples/cub/CMakeLists.txt]: Done.

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2740
* https://travis-ci.org/ingenue/hunter/builds/300191963?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Uploading:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.2748
* https://travis-ci.org/ingenue/hunter/builds/300368470?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Follow-up: https://github.com/ruslo/hunter/commit/1b647e09f1d78b2553e4c38a9e5070e4f58d84a9

Released: https://github.com/ruslo/hunter/releases/tag/v0.19.148

**Comment by headupinclouds** [comment]
Thanks.

**Pull request merged**